elasticsearch/linux_distributions/opendistro-start.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+ES_HOME=`dirname $(realpath $0)`; cd $ES_HOME
+ES_KNN_LIB_DIR=$ES_HOME/plugins/opendistro-knn/knn-lib
+
+function isDarwin {
+    echo "$OSTYPE" | grep -qi "darwin"
+}
+
```

Comment:
Do we even need to write a function for one line code?

Reply:
This can be removed.

Reply:
Functions are cheap, why not use them to make the code easier to understand?

Function names help to convey the intent. `if isDarwin` is much easier than read and parsing `if echo "$OSTYPE" | grep -qi "darwin";`